prover/src/lib.rs

```diff
@@ -60,6 +60,7 @@ pub use winter_prover::{Proof, crypto::MerkleTree as MerkleTreeVC};
 pub fn prove(
     program: &Program,
     stack_inputs: StackInputs,
+    advice_inputs: AdviceInputs,
```

Comment:
`AdviceInputs` and `AdviceProvider` are now exactly the same, right? Is there a reason to keep `AdviceInputs` then?

It's pretty much used as an `AdviceProvider` builder - but without any "building complexity" to take care of. And `AdviceProvider` has mutable methods anyways, which kind of defeats the purpose of the builder. So I think it'd be cleaner to just get rid of `AdviceInputs`?

Reply:
They are indeed the same struct, and could likely be "unified" in some way. 

One blocker at the moment is the stack whose order is reversed in the `AdviceProvider`. 

I'd suggest doing this in a follow-up PR though to minimize the foot-print of this one.

Reply:
#1943